app/assets/javascripts/components/tag_manager.es6.jsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * @prop tags - tags
+ * @prop close_modal_handler - close modal
+ */
+
+class TagManager extends React.Component {
+  
+  constructor(props) {
+    super(props);
+    this._handleKeyPress = this._handleKeyPress.bind(this);
+    this._removeTag = this._removeTag.bind(this);
+    this._success = this._success.bind(this);
+    this._successfulSave = this._successfulSave.bind(this);
+    this.state = {
+      tags: this.props.tags
+    };
+  }
+
+  _successfulSave() {
+    toastr.options.positionClass = 'toast-bottom-right';
+    toastr.success("Save successful!");
+    window.location = "/";
+  }
+
+  _success(msg) {
+    const tagFields = {tags: this.state.tags}
+    APIRequester.put(`/admin_tags/save`, tagFields, this._successfulSave);
+    this.props.close_modal_handler();
+  }
+
+  _handleKeyPress(e) {
+    if (e.key == 'Enter') {
+      e.preventDefault();
+      var newTags = this.state.tags.slice();
+      newTags.push(e.currentTarget.value);
+      this.setState({tags: newTags});
+      e.currentTarget.value = "";
+    }
+  }
+
+  _removeTag(e) {
+    e.preventDefault();
+    var newTags = this.state.tags.filter(function(x) { return x != $(e.target).val() })
```

Comment:
Delete isn't working - I think this should be e.currentTarget.value instead of $(e.target).val()